evm/vm/forks/sharding/vmc_handler.py

```diff
@@ -166,6 +198,67 @@ def fetch_candidate_head(self, shard_id):
         shard_tracker = self.get_shard_tracker(shard_id)
         return shard_tracker.fetch_candidate_head()
 
+    def memoized_fetch_and_verify_collation(self, collation_hash):
```

Comment:
I'm not sure the `memoize_` prefix is needed here.  The fact that it's memoized is an implementation detail.

Reply:
You're right.  Thank you for pointing this out.
Is it okay that I rename this method to the same name as the function `fetch_and_verify_collation` outside?
Or do you have any suggestion on this? thank you:)